### PR TITLE
newdevice: Added support for Terra sti410C

### PR DESCRIPTION
**Pull request by maciejkaczkowski**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: html/images/os/terra.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!-- Created with Inkscape (http://www.inkscape.org/) -->
```
Comment (murrant) [inline html/images/os/terra.svg]: SVG size can be cut down by 1/3, you can trim it down with: https://jakearchibald.github.io/svgomg/

File: includes/definitions/sti410c.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+os: sti410c
```
Comment (laf) [inline includes/definitions/sti410c.yaml]: Is this OS name too specific to one model? Ideally an OS covers a range of models in a product line.

File: includes/polling/os/sti410c.inc.php [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+$version = trim(snmp_get($device, ".1.3.6.1.4.1.30631.1.9.4.1.0", "-Oqv"), '"');
```
Comment (laf) [inline includes/polling/os/sti410c.inc.php]: Please convert these two snmp_get calls to snmp_multi_get_oid()

File: includes/definitions/terra.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+os: terra
```
Comment (laf) [inline includes/definitions/terra.yaml]: That's one extreme to the other :)

Does this cover all terra devices?

Comment (maciejkaczkowski) [inline includes/definitions/terra.yaml]: It covers channel processor devices (DVB-T to IP, DVB-S to DVB-C etc, index 5-10 and 14-18).
All have custom embbeded OS (different for each model), I made tests only for sti410 (9) and sti440 (18). Each device has its own MIBs available after purchase. Only version and serial number are under the same OID.

It doesn't cover products with index:
1 - OD120 - optical receiver - discontinued
2 - MO411 - optical transmitter (other purpose)
3 - MO4x8 - optical transmitter (other purpose - not DVB processor)
4 - UC412 - can't find any info
11 - S2C16 - Similar GUI to sti410, should also work
12 - OT001M - can't find any info
13 - S2C16P - Similar GUI to sti410, should also work

File: includes/discovery/processors/terra.inc.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+if ($device["os"] === "terra") {
+    $descr = "Processor";
+
+    $query = array(
+       array("sti410C", ".1.3.6.1.4.1.30631.1.9.1.1.3.0"),
+       array("sti440",  ".1.3.6.1.4.1.30631.1.18.1.326.3.0")
+    );
+
+    $sysDescr = strlen($device["sysDescr"]) == 0 ? snmp_get($device, ".1.3.6.1.2.1.1.1.0", "-Ovq") : $device["sysDescr"];
```
Comment (laf) [inline includes/discovery/processors/terra.inc.php]: What's the reason for doing this? $device['sysDescr'] should already have what you need.

File: includes/discovery/sensors/temperature/terra.inc.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+if ($device["os"] === "terra") {
+    $query = array(
+        array("sti410C", ".1.3.6.1.4.1.30631.1.9.1.1.4.0"),
+        array("sti440",  ".1.3.6.1.4.1.30631.1.18.1.326.4.0")
+    );
+
+    $sysDescr = strlen($device["sysDescr"]) == 0 ? snmp_get($device, ".1.3.6.1.2.1.1.1.0", "-Ovq") : $device["sysDescr"];
```
Comment (laf) [inline includes/discovery/sensors/temperature/terra.inc.php]: Same here.

File: includes/discovery/sensors/voltage/terra.inc.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+if ($device["os"] === "terra") {
+    $query = array(
+        array("sti410C", ".1.3.6.1.4.1.30631.1.9.1.1.5.0"),
+        array("sti440",  ".1.3.6.1.4.1.30631.1.18.1.326.5.0")
+    );
+
+    $sysDescr = strlen($device["sysDescr"]) == 0 ? snmp_get($device, ".1.3.6.1.2.1.1.1.0", "-Ovq") : $device["sysDescr"];
```
Comment (laf) [inline includes/discovery/sensors/voltage/terra.inc.php]: Same here.

File: includes/discovery/processors/terra.inc.php (hunk 5 above)
Comment (maciejkaczkowski) [inline includes/discovery/processors/terra.inc.php]: $device["sysDescr"] is empty after adding and first discovery (at cron: /opt/librenms/discovery.php -h new). sysDescr is available only after manual discovery or after /opt/librenms/discovery.php -h all. Without that line first graphs are shown only after the discovery.php -h all (after a few hours in the worst case).

Comment (murrant) [inline includes/discovery/processors/terra.inc.php]: Is your install up to date?  This should have been fixed recently.

Comment (maciejkaczkowski) [inline includes/discovery/processors/terra.inc.php]: Created and tested on acaa29c583049ba31937e40e2b339bbc6851dab7 (5 may 2017).

Comment (laf) [inline includes/discovery/processors/terra.inc.php]: I've just tested this and adding a device does populate the sysDescr value straight away.

**Comment by maciejkaczkowski** [comment]
Sorry for the last commit, it should be separate pull request - I don't know how to remove auto commit to current thread.

**Comment by laf** [comment]
Thanks for the pull request.

I reverted your last commit by the way as it was for another device, can you submit that as a different pull request please?

**Comment by maciejkaczkowski** [comment]
Ok, Thank you. I will make few updates and create another request.

**Pull request merged**